python/src/etos_api/library/validator.py

```diff
@@ -191,6 +193,13 @@ async def validate(self, test_suite_url):
                         test_runners.add(constraint.value)
             docker = Docker()
             for test_runner in test_runners:
+                for _ in range(3):
+                    result = await docker.pull(test_runner)
```

Comment:
Is a complete pull necessary here? Can it be avoided?

Reply:
IIRC there is no other way to verify that the image is available for download. @t-persson you're the original author, do you remember?

Reply:
Hrm, looking at this more closely that shouldn't be `docker.pull()` at all. The `Docker()` class doesn't even have a `pull()` method. This is most likely a CoPilot hallucination that slipped past my weary eyes.

Reply:
That also means that there is no pulling of the actual image.

Reply:
fixed